input/postgres/server_stats.go

```diff
@@ -21,12 +21,64 @@ SELECT
 FROM pg_catalog.pg_control_checkpoint()
 `
 
+const xminHorizonSQL string = `
+WITH slots AS (
+	SELECT
+		xmin,
+		catalog_xmin
```

Comment:
Don't we need to query for catalog_xmin separately? (since technically catalog_xmin can be further behind than xmin, and so we need the sort order to be specific to the value we're getting)

i.e. not have this CTE, but instead just query `pg_replication_slots` twice (once ordered by age(xmin) for getting xmin, the other time ordered by age(catalog_xmin) for getting catalog_xmin)

Reply:
Very good point, thanks for pointing this out! Similar to this, I should also tweak the `backend_xmin` part to actually return either `backend_xmin` or `backend_xid`, whatever the smaller (instead of returning backend_xmin, which can be null there if the `backend_xid` is the oldest one). I'll tweak 👍